pkg/sources/forward/data_forward_test.go

```diff
@@ -845,6 +849,10 @@ func (f mySourceForwardTest) ApplyTransform(ctx context.Context, message *isb.Re
 type TestSourceWatermarkPublisher struct {
 }
 
+func (p TestSourceWatermarkPublisher) PublishIdleWatermarks(t time.Time) {
+	// PublishIdleWatermarks is not tested in data_forwarder_test.go
```

Comment:
can we add unit tests and one e2e test for source idling? can be in a separate PR.

Reply:
yaa, I'm working on e2e test